### DISCUSSION TITLE
Unable to update the frontend

### DISCUSSION THREAD

**Discussion by pratikkumar018** [espnet/espnet#4077] [Q&A] [answered]
I am trying to use this recipe(https://github.com/espnet/espnet/blob/master/egs2/librispeech/asr1/conf/tuning/train_asr_conformer7_hubert_ll60k_large.yaml) with the latest espnet . But also trying to update the frontend. So removed freeze_param: ["frontend.upstream"]. I noticed that my overall model is updating but the frontend part is still the same. I tried printing the grads & found it to be None for the frontend parameters. Any help on this?

**Comment by sw005320** [comment]
@simpleoier, could you answer it?

**Comment by simpleoier** [comment, marked as answer]
@pratikkumar018 If you check the file `espnet2/asr/frontend/s3prl.py`, you will notice in the forward function, there is this [line](https://github.com/espnet/espnet/blob/a3e1543e96c1088bfed846d5c68c6f444a55aa75/espnet2/asr/frontend/s3prl.py#L126) preventing the updates. I'm considering to remove this line soon.

**Reply by pratikkumar018** [reply]
Thanks for quick reply. Model is updating now.
